Replicator Panics During Cluster Scale-In Operations When The Prioritized Scaling Metric Is Disk

**Description**

When Replicator has identified the prioritized scaling metric as `ScalingMetricDisk` and attempts to initiate a cluster scale-in operation, a panic is triggered while attempting to identify the least-allocated node. 

The fundamental cause of the panic is the lack of support for `ScalingMetricDisk` in the `LeastAllocatedNode` method while the `MostUtilizedResource` method implements this support. This results in no node ID being set and passing a `nil` to the Nomad API while attempting to retrieve node details. 

**Describe the results you expected:**

If Replicator determines the most utilized resource is the disk space, it should be able to successfully initiate a cluster scaling operation in any direction without erroring out. 

**Describe the results you received:**

Replicator fails to identify the least allocated node when initiating a cluster scale-in operation and therefore, cannot affect the cluster scaling operation.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Replicator version:** 

```
Replicator v1.1.0-dev
```

**Output of `consul version`:** 

```
Consul v1.0.0
```

**Output of `nomad version`:** 

```
Nomad v0.7.0
```

**Additional environment details:**

**Debug log outputs from replicator:**
```
[DEBUG] client/nomad: Node 28738238-b5b8-c889-261a-af86c991adb0 will be excluded when calculating eligible worker pool nodes to be removed
[ERROR] client/nomad: unable to determine nomad node IP address: Unexpected response code: 500 (3 error(s) occurred:

* RPC failed to server 10.188.33.40:4647: rpc error: missing node ID
* RPC failed to server 10.188.33.69:4647: rpc error: rpc error: missing node ID
* RPC failed to server 10.188.32.178:4647: rpc error: rpc error: missing node ID)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0xa897ea]

goroutine 249 [running]:
github.com/elsevier-core-engineering/replicator/client.(*nomadClient).LeastAllocatedNode(0xc420188060, 0xc420194c00, 0xc4200191d0, 0x24, 0x7, 0x1, 0x0, 0xc420016e80)
	/home/travis/gopath/src/github.com/elsevier-core-engineering/replicator/client/nomad.go:152 +0x3aa
github.com/elsevier-core-engineering/replicator/replicator.(*Runner).workerPoolScaling(0xc420182560, 0xc420190480, 0x21, 0xc42018e940, 0xc420184990, 0xc420442870)
	/home/travis/gopath/src/github.com/elsevier-core-engineering/replicator/replicator/cluster_scaling.go:208 +0xa2b
created by github.com/elsevier-core-engineering/replicator/replicator.(*Runner).asyncClusterScaling
	/home/travis/gopath/src/github.com/elsevier-core-engineering/replicator/replicator/cluster_scaling.go:92 +0x11e
```

--- a/client/nomad.go
+++ b/client/nomad.go
@@ -131,13 +131,18 @@ func (c *nomadClient) LeastAllocatedNode(capacity *structs.ClusterCapacity,
 		switch capacity.ScalingMetric.Type {
 		case ScalingMetricProcessor:
 			if (lowest == 0) || (alloc.UsedCapacity.CPUPercent < lowest) {
-				nodeID = alloc.NodeID
 				lowest = alloc.UsedCapacity.CPUPercent
+				nodeID = alloc.NodeID
 			}
 		case ScalingMetricMemory:
 			if (lowest == 0) || (alloc.UsedCapacity.MemoryPercent < lowest) {
-				nodeID = alloc.NodeID
 				lowest = alloc.UsedCapacity.MemoryPercent
+				nodeID = alloc.NodeID
+			}
+		case ScalingMetricDisk:
+			if (lowest == 0) || (alloc.UsedCapacity.DiskPercent < lowest) {
+				lowest = alloc.UsedCapacity.DiskPercent
+				nodeID = alloc.NodeID
 			}
 		case ScalingMetricNone:
 			nodeID = alloc.NodeID

